### PR TITLE
Eslint, prettier and GitHub action

### PR DESCRIPTION
**Pull request by Nathan-SL**
- Introduces GitHub action to handle formatting and linting on pushes to branches `master` and `plugin` and PRs.
- Resolves a conflict between `prettier` and `eslint` by running the command `npm run format` offered by [plugin](https://github.com/prettier/prettier-eslint-cli). 
    - Main issue: The example used [here](https://prettier.io/docs/en/options.html#quotes) shows that prettier changes quotes (single/double) based on the number of quotes in the string.
    - PS: Using `singleQuote: true` in `.prettierc` wont be the solution.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+hardhat.config.ts
```
Comment (FabijanC) [inline .eslintignore:1]: Why?

File: .eslintrc.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   },
   "plugins": ["@typescript-eslint"],
   "rules": {
+    "@typescript-eslint/ban-ts-comment": "off", // Should be enabled after TODOs are resolved
```
Comment (FabijanC) [inline .eslintrc.json:14]: Instead of having it off, you can just temporarily suppress TODO errors in that file

File: .prettierignore [hunk 3]
```diff
@@ -0,0 +1 @@
+hardhat.config.ts
```
Comment (FabijanC) [inline .prettierignore:1]: Why?

**Review by FabijanC — CHANGES_REQUESTED** [review]
I don't like the fact that we would have to take care of the order (first format, then lint). You can freely introduce the mentioned plugin if it resolves this issue.

File: .eslintignore (hunk 1 above)
Comment (Nathan-SL) [reply .eslintignore:1]: The config file has `require("dotenv").config();` which the eslint considers as an error.

File: .eslintrc.json (hunk 2 above)
Comment (Nathan-SL) [reply .eslintrc.json:14]: Actually, that comment is somehow misleading. The line is to turn off the rule for `@ts-ignore` which if we don't turn off gives us an error from the linter. The part for the TODO errors is the rule `no-warning-comments` which is also commented on line 50

File: .eslintignore (hunk 1 above)
Comment (FabijanC) [reply .eslintignore:1]: Then change the require to import.

File: .eslintrc.json (hunk 2 above)
Comment (FabijanC) [reply .eslintrc.json:14]: Then suppress ts-ignore in that file

Comment (Nathan-SL) [reply .eslintrc.json]: https://github.com/Shard-Labs/starknet-hardhat-example/pull/74/commits/9113384642c706ac1504067dac078d89c7a0857c

File: .eslintignore (hunk 1 above)
Comment (Nathan-SL) [reply .eslintignore]: https://github.com/Shard-Labs/starknet-hardhat-example/pull/74/commits/14dca5d1bc47a34696a306f36f335d59a0a50921

File: .prettierignore (hunk 3 above)
Comment (Nathan-SL) [reply .prettierignore]: > I don't like the fact that we would have to take care of the order (first format, then lint). You can freely introduce the mentioned plugin if it resolves this issue.

The plugin mentioned in the old PR description doesn't offer a CLI out of the box, [this](https://github.com/Shard-Labs/starknet-hardhat-example/pull/74/commits/14dca5d1bc47a34696a306f36f335d59a0a50921) introduce a new one with the same functionality.

**Review by FabijanC — APPROVED** [review]

**Pull request merged**